backend-demo/src/screencast.rs

```diff
@@ -0,0 +1,82 @@
+use ashpd::{
+    backend::{
+        request::RequestImpl,
+        screencast::{
+            CreateSessionOptions, ScreencastImpl, SelectSourcesOptions, SelectSourcesResponse,
+            StartCastOptions,
+        },
+        session::{CreateSessionResponse, SessionImpl},
+        Result,
+    },
+    desktop::{
+        screencast::{CursorMode, SourceType, StreamBuilder, Streams, StreamsBuilder},
+        HandleToken,
+    },
+    enumflags2::BitFlags,
+    AppID, WindowIdentifierType,
+};
+use async_trait::async_trait;
+
+#[derive(Default)]
+pub struct Screencast {}
+
+#[async_trait]
+impl RequestImpl for Screencast {
+    async fn close(&self, token: HandleToken) {
+        tracing::debug!("IN Close(): {token}");
+    }
+}
+
+#[async_trait]
+impl ScreencastImpl for Screencast {
+    fn available_source_types(&self) -> BitFlags<SourceType> {
+        SourceType::Monitor | SourceType::Window
+    }
+
+    fn available_cursor_mode(&self) -> BitFlags<CursorMode> {
+        CursorMode::Hidden | CursorMode::Embedded | CursorMode::Metadata
+    }
+
+    async fn create_session(
+        &self,
+        _handle_token: HandleToken,
```

Comment:
i would name all the _handle_token to just token and session_handle_token to just session_token